### PR TITLE
Update azure standard tests to use macOS-15 and python3.12

### PR DESCRIPTION
**Pull request by ohemorange**
macOS-12 is [being deprecated](https://github.com/actions/runner-images/issues/10721) on Azure, so update to the latest available version.

We use python3.12 in our snap, so that should be the default version used for the standard tests (with 3.8 relegated to oldest and extended tests, at least for now).

### CODE REVIEW

File: .azure-pipelines/templates/jobs/standard-tests-jobs.yml [hunk 1]
```diff
@@ -4,26 +4,24 @@ jobs:
...
+          IMAGE_NAME: macOS-15
           TOXENV: cover
           # As of pip 23.1.0, builds started failing on macOS unless this flag was set.
           # See https://github.com/certbot/certbot/pull/9717#issuecomment-1610861794.
           PIP_USE_PEP517: "true"
         macos-cover:
-          IMAGE_NAME: macOS-13
+          IMAGE_NAME: macOS-15
           TOXENV: cover
           # See explanation under macos-py38-cover.
```
Comment (bmw) [inline .azure-pipelines/templates/jobs/standard-tests-jobs.yml:16]: we should update this comment too if we're renaming the test

i believe these two macOS tests are now identical though with the removal of setting `PYTHON_VERSION: 3.8` in the previous test. this is not at all clear, but i believe the idea here was we run macOS coverage tests once with the oldest version of python we support and once with the newest

i think my personal preference is to keep that pattern (and iif you're feeling extra motivated refactor things or add comments to make that more clear), but we could also just delete one of the two tests. i'd approve either approach

File: .azure-pipelines/templates/jobs/standard-tests-jobs.yml [hunk 2]
```diff
@@ -4,26 +4,24 @@ jobs:
...
           TOXENV: oldest
-        linux-py38:
+        linux-py312:
           IMAGE_NAME: ubuntu-22.04
-          PYTHON_VERSION: 3.8
-          TOXENV: py38
+          TOXENV: py312
         linux-cover:
           IMAGE_NAME: ubuntu-22.04
           TOXENV: cover
```
Comment (bmw) [inline .azure-pipelines/templates/jobs/standard-tests-jobs.yml:27]: i believe we're using the same pattern here testing on linux with the oldest and newest python we support, except we're only measuring/testing coverage once. as of now in this PR, these tests aren't identical, but i believe `linux-cover` is a superset of the tests run in `linux-py312`

**Review by bmw — CHANGES_REQUESTED** [review]
thanks for saving us from potential trouble on release day

File: .azure-pipelines/templates/jobs/standard-tests-jobs.yml (hunk 1 above)
Comment (ohemorange) [reply .azure-pipelines/templates/jobs/standard-tests-jobs.yml:16]: oh, missed that, I'll change these back. still going to keep it on the updated mac version though unless there's some reason not to.

**Review by bmw — APPROVED** [review]
lgtm!

**Pull request merged**